### PR TITLE
Update free-courses-en.md

### PR DESCRIPTION
**Pull request by Devadeut**
added new link for data structures and algorithm course

## What does this PR do?
Add resource(s) 

## For resources
ECS 36C is a Course for data structures and algorithms from [Joël Porquet-Lupine] currently an Assistant Professor of Teaching at UC Davis.

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [ ] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: courses/free-courses-en.md [hunk 1]
```diff
@@ -190,6 +190,7 @@
 * [Stanford University Algorithms: Design and Analysis, Part 1](https://online.stanford.edu/courses/soe-ycsalgorithms1-algorithms-design-and-analysis-part-1)
 * [Stanford University Algorithms: Design and Analysis, Part 2](https://online.stanford.edu/courses/soe-ycs0001-algorithms-design-and-analysis-part-2)
 * [Strivers A2Z-DSA Course \| DSA Playlist \| Placements](https://www.youtube.com/playlist?list=PLgUwDviBIf0oF6QL8m22w1hIDC1vJ_BHz) - take U forward
+* [The Data Structures, Algorithms, and Programming](https://lupteach.gitlab.io/courses/ucd-ecs36c/online/) -Prof. Porquet-Lupine, UC Davis
```
Comment (eshellman) [inline courses/free-courses-en.md:193]:
```suggestion
* [ECS 36C: Data Structures, Algorithms, and Programming](https://lupteach.gitlab.io/courses/ucd-ecs36c/online/) - Joël Porquet-Lupine
```
Don't invent titles! Use the title given by the resource.
No honorifics

Comment (Devadeut) [reply courses/free-courses-en.md:193]: Okay, sure.

**Comment by Devadeut** [comment]
@eshellman I made the necessary changes. Kindly review it and Merge.

File: courses/free-courses-en.md [hunk 2]
```diff
@@ -190,6 +191,7 @@
...
```
Comment (eshellman) [inline courses/free-courses-en.md:194]: no duplicates! pick a single category.

**Comment by Devadeut** [comment]
@eshellman removed the duplicates.

**Pull request merged**